src/Infrastructure.VS.UnitTests/SolutionInfoProviderTests.cs

```diff
@@ -64,19 +77,64 @@ public async Task Get_ReturnsExpectedValue(string solutionNameToReturn)
         }
 
         [TestMethod]
-        public async Task Get_ServiceCalledOnBackgroundThread()
+        [DataRow(null)]
+        [DataRow("a value")]
+        public void Get_ReturnsExpectedValue(string solutionNameToReturn)
+        {
+            var solution = CreateIVsSolution(solutionNameToReturn);
+            var serviceProvider = CreateServiceProviderWithSolution(solution.Object);
+
+            var testSubject = CreateTestSubject(serviceProvider.Object);
+
+            var actual = testSubject.GetFullSolutionFilePath();
+            actual.Should().Be(solutionNameToReturn);
+        }
+
+        [TestMethod]
+        public async Task GetAsync_ServiceCalledOnUIThread()
         {
             var calls = new List<string>();
 
             var solution = CreateIVsSolution("any", () => calls.Add("GetSolutionName"));
             var serviceProvider = CreateServiceProviderWithSolution(solution.Object, () => calls.Add("GetService"));
-            var threadHandling = CreateThreadHandlingWithRunOnUICallback(() => calls.Add("RunOnUIThread"));
+
+            var threadHandling = new Mock<IThreadHandling>();
+            threadHandling.Setup(x => x.RunOnUIThreadAsync(It.IsAny<Action>()))
+                .Callback<Action>(productOperation =>
+                {
+                    calls.Add("RunOnUIThreadAsync");
+                    productOperation.Invoke();
+                });
+
 
             var testSubject = CreateTestSubject(serviceProvider.Object, threadHandling.Object);
 
             var actual = await testSubject.GetFullSolutionFilePathAsync();
 
-            calls.Should().ContainInOrder("RunOnUIThread", "GetService", "GetSolutionName");
+            calls.Should().ContainInOrder("RunOnUIThreadAsync", "GetService", "GetSolutionName");
+        }
+
+        [TestMethod]
+        public void Get_ServiceCalledOnUIThread()
```

Comment:
Copy of the `GetAsync_ServiceCalledOnUIThread` test, modified to test the synchronous method.